### PR TITLE
removed statefulset implementation of prombench tests

### PR DESCRIPTION
**Pull request by geekodour**
closes #280 
closes #278 
closes #270 

Signed-off-by: Hrishikesh Barman <hrishikeshbman@gmail.com>

cc @krasi-georgiev

### CODE REVIEW

**Comment by geekodour** [comment]
@krasi-georgiev 
github workflow file to start, cancel and restart tests:

https://github.com/prometheus-community/prometheus/blob/master/.github/workflows/prombench.yml

**Comment by krasi-georgiev** [comment]
why is it needed to have 2 workflows ? for example `pre_benchmark_restart` and `benchmark_restart`

File: prombench/design.svg [hunk 1]
```diff
@@ -1,3 +1,3 @@
...
...(51 , 51 , 51) ; text-align: center">GitHub Actions </span></div><div style="text-align: left"><span style="color: rgb(51 , 51 , 51) ; text-align: center">starts a new test</span><br /></div></div></div></foreignObject><text x="44" y="20" fill="#000000" text-anchor="middle" font-size="12px" font-family="Helvetica">[Not supported by viewer]</text></switch></g><rect x="155" y="479.5" width="160" height="50" rx="7.5" ry="7.5" fill="#dae8fc" stroke="#6c8ebf" pointer-events="none"/><g transform="translate(159.5,487.5)"><switch><foreignObject style="overflow:visible;" pointer-events="all" width="151" height="34" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; font-size: 15px; font-family: Helvetica; color: rgb(0, 0, 0); line-height: 1.2; vertical-align: top; width: 152px; white-space: nowrap; overflow-wrap: normal; text-align: center;"><div xmlns="http://www.w3.org/1999/xhtml" style="display:inline-block;text-align:inherit;text-decoration:inherit;white-space:normal;"><font>Loki<br /><font style="font-size: 10px">(promtail will push metrics to Loki)</font></font></div></div></foreignObject><text x=
```
Comment (krasi-georgiev) [inline prombench/design.svg]: the line wrapping is not right on some blocks
https://github.com/prometheus/prombench/tree/d6783184d6e09887d261ad6ece5481b1b3a2fbfe/prombench

**Review by krasi-georgiev — COMMENTED** [review]
.

**Comment by geekodour** [comment]
@krasi-georgiev 

> why is it needed to have 2 workflows ? for example `pre_benchmark_restart` and `benchmark_restart` 

if conditional for `steps` is not implemented as mentioned by github staff in this [ticket](https://github.community/t5/GitHub-Actions/Github-Actions-Contexts-and-steps-if-conditionals-are-unusable/m-p/29669/highlight/true#M8294)

So we need to run `pre_benchmark_*` where we validate the comment, then we run `benchmark_*` where we extract the arg and run prombench.

we use [`job.needs`](https://help.github.com/en/articles/workflow-syntax-for-github-actions#jobsjob_idneeds) to use `pre_benchmark_*` in `benchmark_*`

File: prombench/design.svg (hunk 1 above)
Comment (geekodour) [reply prombench/design.svg]: okay will update this

**Comment by krasi-georgiev** [comment]
Why can't comment validation and extracting the args happen in the same workflow?

**Comment by geekodour** [comment]
> Why can't comment validation and extracting the args happen in the same workflow?

**In short:**
If we put comment validation and extracting the args in the same job, the failure comment will be sent even if comment validation failed.

**long:**
there are currently `two` workflows in [prometheus-community/prometheus/.github/workflow](https://github.com/prometheus-community/prometheus/tree/master/.github/workflows):
- `funcbench.yml`
- `prombench.yml`

name of the `prombench.yml` `workflow` is `Prombench Workflow` 
```
on: issue_comment
name: Prombench Workflow
.
.
.
```

`Prombench Workflow` contains the following `job`(s) and corresponding `step`(s):

---
- `pre_benchmark_start`
  - `validate_comment`
- `benchmark_start`
  - `extract_arguments`
  - `make_deploy`
  - `fail_notification`
---
- `pre_benchmark_restart`
  - `validate_comment`
- `benchmark_restart`
  - `extract_arguments`
  - `make_restart`
  - `fail_notification`
---
- `pre_benchmark_cancel`
  - `validate_comment`
- `benchmark_cancel`
  - `extract_arguments`
  - `make_clean`
  - `fail_notification`

We need to stop the `job` if the comment is not valid, and we need to post a comment if any of the `make_*` fails.

This could be possible if `step` had `if` conditionals, but [from this ticket](https://github.community/t5/GitHub-Actions/Github-Actions-Contexts-and-steps-if-conditionals-are-unusable/m-p/29669/highlight/true#M8294) , GitHub staff informs that it's not implemented yet. only `if: failure()` and `if: success()` works, which we're making use of in the workflow

when `pre_benchmark_*/validate_comment` is a `success`, comments are extracted but we're not making use of it.

We use [`job.needs`](https://help.github.com/en/articles/workflow-syntax-for-github-actions#jobsjob_idneeds) to use `pre_benchmark_*` in `benchmark_*`

So `benchmark_*` `job`(s) are only run when corresponding `pre_benchmark_*` are successful. 

Now we have three `steps` inside the `benchmark_*` jobs.
- `extract_arguments` : this will be successful if `validate_comment` was successful
- `make_*` : make use of the arguments extracted by `extract_arguments` and run corresponding `action`
- `fail_notification`: comment using `comment-monitor` if corresponding `make_*` results in a `failure`

**Pull request merged**